test/e2e/specs/editor/blocks/links.spec.js

```diff
@@ -445,11 +445,13 @@ test.describe( 'Links', () => {
 		await pageUtils.pressKeys( 'Enter' );
 
 		const linkPopover = LinkUtils.getLinkPopover();
+		await expect( linkPopover ).toBeVisible();
+		// Close the link control to return the caret to the canvas
+		await pageUtils.pressKeys( 'Escape' );
 
 		// Deselect the link text by moving the caret to the end of the line
 		// and the link popover should not be displayed.
 		await pageUtils.pressKeys( 'End' );
-		await expect( linkPopover ).toBeHidden();
```

Comment:
Should we invert this instead? `toBeVisible()`?

Reply:
You'd need to do a little bit more, since the focus is in a different spot. I modified it to match the spirit of what the test was already testing. https://github.com/WordPress/gutenberg/pull/57726/commits/fb8f5a9d96075154ad61b13f5b14265b4574e26d